woocommerce/woocommerce
# Try reducing blocks build size

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR updates the `.distignore` used for building the core release package to exclude other files from the `.distignore` [that was in the blocks repo](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/.distignore).

I've also updated the `build-zip` script within core to set `NODE_ENV` to `production`, as [this affects the generation of source-maps](https://github.com/woocommerce/woocommerce/blob/0cfea936e4ad0e102c89916140d9ee083e5f1bcc/plugins/woocommerce-blocks/webpack.config.js#L34). I had to do this after `pnpm install`, as doing it prior broke several other aspects of the build.





Closes # .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Run the [Build Release Zip workflow](https://github.com/woocommerce/woocommerce/actions/workflows/build-release-zip-file.yml) from this branch (or view this [existing workflow run](https://github.com/woocommerce/woocommerce/actions/runs/7427281984))
2. Compare the release asset to a zip generated in `trunk` to verify that it's of smaller size.
3. Confirm that nothing is missing from the zip file that needs to be there (spot-check that nothing was added to `.distignore` that shouldn't have been)
4. Verify that the zip installs, functions as expected.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 
This PR affects build tooling and a change entry is not required.

</details>
